### PR TITLE
docs: updating ssr docs

### LINKED ISSUE #1811: Improve docs for SSR on tRPC
From @TheoBr - https://clips.twitch.tv/SleepyVibrantGrasshopperDxCat-ZJjzoC-yQHmSmyDE

- [ ] Add this on the `/examples/next-prisma-starter`: https://github.com/trpc/examples-next-prisma-starter-websockets/blob/main/src/pages/_app.tsx#L75
- [ ] Add docs in https://trpc.io/docs/ssr
- [ ] https://github.com/trpc/examples-next-prisma-starter/blob/db7e2c1469170a660a12a45d7c40829ed4bc9509/src/pages/_app.tsx#L28-L44
- [ ] Make sure you don't use `getServerSideProps` or `getStaticProps` anywhere in the app  (#596)

### PR DESCRIPTION
**Pull request by rhenriquez28**
Closes #1811 

## 🎯 Changes

What changes are made in this PR? Is it a feature or a bug fix?
- Added client's headers forwarding example
- Removed "Caveats" in favor of "FAQ"

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [ ] If necessary, I have added documentation related to the changes made.
- [ ] I have added or updated the tests related to the changes made.

### CODE REVIEW

**Comment by rhenriquez28** [comment]
Should I add these changes to the v9 docs?

**Review by juliusmarminge — APPROVED** [review]
> Should I add these changes to the v9 docs?

I think that would be nice since it applies there too.

File: www/versioned_docs/version-9.x/nextjs/ssr.md [hunk 1]
```diff
@@ -47,16 +47,47 @@ export default withTRPC<AppRouter>({
...
   },
   ssr: true,
 })(MyApp);
 ```
 
-## Caveats
+
+## FAQ
+
+### Q: Why do I need to forward the client's headers to the server manually? Why doesn't tRPC automatically does that for me?
```
Comment (juliusmarminge) [inline www/versioned_docs/version-9.x/nextjs/ssr.md:83]:
```suggestion
### Q: Why do I need to forward the client's headers to the server manually? Why doesn't tRPC automatically do that for me?
```

File: www/docs/nextjs/ssr.md [hunk 2]
```diff
@@ -62,6 +83,17 @@ const MyApp: AppType = ({ Component, pageProps }: AppProps) => {
 export default trpc.withTRPC(MyApp);
 ```
 
-## Caveats
+## FAQ
+
+### Q: Why do I need to forward the client's headers to the server manually? Why doesn't tRPC automatically does that for me?
```
Comment (juliusmarminge) [inline www/docs/nextjs/ssr.md:88]:
```suggestion
### Q: Why do I need to forward the client's headers to the server manually? Why doesn't tRPC automatically do that for me?
```

**Review by juliusmarminge — APPROVED** [review]

**Review by KATT — APPROVED** [review]

**Pull request merged**